WordPress/gutenberg
# DropdownMenu: fix icon style when dashicon is used

Fix #29376, #40138 

## What?
This PR fixes a problem where the controls property of the `DropdownMenu` component doesn't apply the correct style when **dashicon** is specified.

## Why?
In #29376 and #40138, you can see sample code using `registerFormatType` and `RichTextToolbarButton`. These codes add new formatting capabilities to rich text and are added in the same dropdown as the core formatting.

![format](https://user-images.githubusercontent.com/54422211/186406878-bdec8b80-39e5-43a7-943e-0486cb7d17d3.png)

Also, the icon property of the `RichTextToolbarButton` is internally passed to the `Button` component within the `DropdownMenu` component.

This button icon style is applied correctly for SVG icons such as `@wordpress/icons`, but not for dashicons that are output as span tags. Therefore, it is not visible that the format is applied (i.e., the icon is active).

At the same time, there is a difference in the default style when the SVG icon and the dash icon are used at the same time.

## How?
I have added selectors so that styles for SVG icons also apply to dashicons. At the same time, ~~I have added a style to center it because the dashicon font is displayed as a pseudo-element.~~ I have removed unnecessary styles.

## Testing Instructions
- Open the side / post editor.
- Enter text using paragraph blocks, etc.
- Run the following code from the browser console to apply the custom formatting.

```javascript
wp.richText.registerFormatType( 'test/custom', {
  title: 'Custom Format',
  tagName: 'span',
  className: 'custom',
  edit: ( { isActive, value, onChange } ) => wp.element.createElement(
    wp.blockEditor.RichTextToolbarButton,
    {
       icon: 'editor-code',
       isActive,
       title: "Custom Format",
       onClick: () => onChange( wp.richText.toggleFormat( value, { type: 'test/custom' } ) ),
    },
    null
  ),
});
```

- Apply the added custom format and the format of the core.
- Confirm that the custom format icon is now active and in the same style as the core icon.

## Screenshots or screencast

| On Trunk | On this branch |
| ---- | ---- | 
|  ![trunk](https://user-images.githubusercontent.com/54422211/186408363-b0a8e768-e0ad-47a6-8052-816dab34409d.png)  |  ![branch](https://user-images.githubusercontent.com/54422211/186408334-078584f4-ea42-41cb-91a4-be1b087f6caf.png)  |
